### ISSUE TITLE
Handy generic error: ErrorCore.DEFAULT

### ISSUE THREAD

**Issue by vitovalov** [mobilejazz/MJ-Android-Core#2]
It would be nice to have a mechanism of creating some generic error message instead of ErrorCore.EMPTY which doesn’t add much value. It would be really comfortable to have ErrorCore.DEFAULT to not spend time creating some default error on every project.
There’re times when the error is undefined for now but you still have to cover the error case with something. Otherwise it will be lost and will pass unobserved.

**Comment by aldoborrero** [comment]
Yup, as is an android project we can add a default string and create a helper or something like that in which can retrieve the string value and display a toast for example

-- 
Aldo Borrero
Sent with Airmail

**Comment by joselufo** [comment]
The idea to the **ErrorCore.EMPTY** was for the case that you don't need to send any error but you can to add a **ErrorCore<?>** object because is a library requirement.

Inside the **ErrorCore<?>** class you have two methods:

`public static <T extends RuntimeException> ErrorCore<?> of(T cause, String errorMessage)`
`public static <T extends RuntimeException> ErrorCore<?> of(T cause)`

For example, if you need a simple **ErrorCore<?>** object, you can just call to the "of" method and fill it with a RuntimeException and a string message that you need instead to create your own **ErrorAdapter**

I'm open to new ideas, but I need more information what do you need exactly.

**Comment by vitovalov** [comment]
The thing I didn't like about creating ErrorCore object and calling of method with new RuntimeException everytime, is that your code will be all populated with lots of RuntimeExcetions here and there. IMO RuntimeException name is not really appropriate for this case. 
So:
- I think **of(String errorMessage)** is a must feature missing here

- I suggest creating ErrorCoreException extending RuntimeException at least. The name can be confusing in some places. Or another name if you can come up with.

- Still I would prefer some default static default message object which would save some time in every new project we start. (yes it's little time, but then why is the EMPTY one there? For easiness and comfort. So would be the default one. But up to you..
